### ISSUE TITLE
log scale with zero data should give a better warning

### ISSUE THREAD

**Issue by jowens** [vega/vega#1334]
In Altair, I had a column of data that I plotted on a log scale:

```
sh = Chart(df).mark_point().encode(
    x='Launch:T',
    y=Y('Pixel/unified shader count:Q',
        scale=Scale(type='log'),
        ),
```

and a zero snuck into the column that I didn't know about and the debugging of that took a really really long time. Here's what Altair output looked like:

![image](https://user-images.githubusercontent.com/976970/41696109-2bcc78fa-74c7-11e8-9c92-80a6b21b7717.png)

I guess I would have expected that the zeroes would be silently dropped OR the console would have thrown a warning ... anyway, the Altair output was perplexing. One zero out of several hundred data points made everything plot on a single line.

@jakevdp pointed me to vega-lite to file this issue (https://github.com/altair-viz/altair/issues/953) and @domoritz said send it here (https://github.com/vega/vega-lite/issues/3911).

@domoritz: 
> One thing we could do in Vega-Lite is automatically filter out values less or equal to zero (in https://github.com/vega/vega-lite/blob/master/src/compile/data/filterinvalid.ts) but that seems like it could have all kinds of bad effects.
> 
> Yes, I think this needs to be in Vega itself.

**Comment by jheer** [comment]
This should be doable by adding some error checking after domain construction in vega-encode's Scale transform. If the domain is not strictly positive or negative (excluding zero) we could raise an error or warning.

I'm not yet sure what severity to assign the problem: should this result in an error or warning log message? I'm leaning towards error, but am happy to hear other thoughts! (Note that this corresponds to calling `logger.error`, not `throw new Error(...)`!)

**Comment by jowens** [comment]
I could see all sorts of unforeseen consequences (like @domoritz said) from a full error, so y'all can talk that through, but at least in the nearer term, a warning would probably be useful for debugging.

**Comment by jheer** [comment]
Added a warning log message for zero in log scale domains to the `Scale` transform. Released in vega-encode 3.1.1, will be included in the next Vega release.
